optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -151,34 +144,62 @@ public void solve(DefaultSolverScope<Solution_> solverScope) {
                 phaseScope.setLastCompletedStepScope(stepScope);
             }
         } finally {
-            // If a partition thread throws an Exception, it is propagated here
-            // but the other partition threads won't finish any time soon, so we need to ask them to terminate
+            // 1. In case on of the partition threads threw an Exception, it is propagated here
+            // but the other partition threads are not aware of the failure and may continue solving for a long time,
+            // so we need to ask them to terminate. In case no exception was thrown, this does nothing.
             childThreadPlumbingTermination.terminateChildren();
-            try {
-                // First wait for solvers to terminate voluntarily (because we have just issued children termination)
-                if (!threadPoolExecutor.awaitTermination(1, TimeUnit.SECONDS)) {
-                    // Some solvers refused to complete. Busy threads will be interrupted in the finally block
-                    logger.warn("{}Partitioned Search threadPoolExecutor didn't terminate within timeout (1 second).",
-                            logIndentation);
+
+            if (Thread.currentThread().isInterrupted()) {
+                // 2a. If the current thread is interrupted, propagate interrupt signal to children by initiating
+                // an abrupt shutdown.
+                executor.shutdownNow();
```

Comment:
I still think it might be worthy to await thread pool termination on this branch. I just don't have the supporting arguments at this moment.

Reply:
Agreed. from shutdownNow() javadoc: "This method does not wait for actively executing tasks to
     * terminate.  Use {@link #awaitTermination awaitTermination} to
     * do that."

Reply:
Please change indeed, I 'll wait for that change to merge

Reply:
Done.